dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -256,9 +261,43 @@ private WorkflowVersion updateDBVersionSourceFilesWithRemoteVersionSourceFiles(W
         for (Validation versionValidation : remoteVersion.getValidations()) {
             existingVersion.addOrUpdateValidation(versionValidation);
         }
+
+        // Setup CheckUrl
+        if (checkUrlLambdaUrl != null) {
+            publicAccessibleUrls(existingVersion, checkUrlLambdaUrl);
+        }
+
         return existingVersion;
     }
 
+    /**
+     * Sets the publicly accessible URL version metadata.
+     *
+     * @param existingVersion   Hibernate initialized version
+     * @param checkUrlLambdaUrl URL of the checkUrl lambda
+     */
+    public static void publicAccessibleUrls(WorkflowVersion existingVersion, String checkUrlLambdaUrl) {
+        existingVersion.getSourceFiles().forEach(sourceFile -> {
+            if (sourceFile.getType().getCategory().equals(DescriptorLanguage.FileTypeCategory.TEST_FILE)) {
+                Optional<Boolean> publicAccessibleUrls = Optional.empty();
+                if (sourceFile.getAbsolutePath().endsWith(".json")) {
+                    publicAccessibleUrls =
+                        CheckUrlHelper.checkTestParameterFile(sourceFile.getContent(), checkUrlLambdaUrl, TestFileType.JSON);
+                } else {
+                    if (sourceFile.getAbsolutePath().endsWith(".yaml") || sourceFile.getAbsolutePath().endsWith(".yml")) {
+                        publicAccessibleUrls = CheckUrlHelper.checkTestParameterFile(sourceFile.getContent(), checkUrlLambdaUrl,
+                            TestFileType.YAML);
+                    }
+                }
+                if (publicAccessibleUrls.isEmpty()) {
```

Comment:
If I'm reading this correctly, won't this set publicAccessibleTestParameterFile based on the last processed test parameter file? For example, if you have two test parameter files

* a.json has all public urls
* b.json does not have all public urls

If b.json is processed last, then `setPublicAccessibleTestParameterFile` will be called last with false. But you would want it to be true, since there is at least one file with all public urls.

Also, and maybe this is a phase 2 thing, it would be nice to know which test parameter files contain all public urls and which ones don't. Should this info instead be stored with the file, or as well?

Reply:
good point. looking at workflow registration dialog and most existing workflows, I've forgotten we even allowed multiple test parameter files